### PR TITLE
Using Multiple SSH Keys for Multiple GitHub Accounts

### LINKED ISSUE #2595: Using Multiple SSH Keys for Multiple GitHub Accounts
We ask that you please be patient as our team works through approving and publishing all articles/tutorials in a timely manner. Allow 1-3 days for a topic to be reviewed and/or approved - allow 3-7 days for an articles to be reviewed and/or published.

Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles to go over.

**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**NOTE: We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

*We're looking for the first 2-3 paragraphs of the article that appropriately summarize what your article will be about.*

### Using Multiple SSH Keys for Multiple GitHub Accounts
Your title should be descriptive of the article/tutorial. Be Specific. *Use keyword research to gain your article higher ranking.*

### Introduction paragraph (2-3 paragraphs):
GitHub is a code hosting platform for collaboration. It enables people to work on projects from anywhere.
Most GitHub users have a problem with managing multiple GitHub accounts. This article aims to let the user understand how to use SSH Keys with multiple GitHub accounts.
The article will answer several questions, it will address what SSH keys are and how to generate them, how to manage SSH keys, and how to use them for multiple GitHub Accounts.

### Key takeaways: 
1. What are SSH Keys
2. How to manage SSH Keys on GitHub Accounts
3. What is SSH config and how to set up one
4. How to use SSH Keys for multiple GitHub Accounts

### References: 
Please list links to any published content/research that you intend to use to support/guide this article.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by LeahKui**
### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**
      

This closes #2595
  
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by LeahKui** [comment]
This fixes #2595

**Comment by LeahKui** [comment]
Hi @srishilesh, the author page and hero image are part of files committed previously

File: content/authors/wangui-leah/index.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Leah is an information technology enthusiast. She is passionate about software development and technical writing. Her areas of interest are Data Science and Machine learning(ML). In addition, she loves playing computer games and adventure.
```
Comment (srishilesh) [inline content/authors/wangui-leah/index.md:1]:
```suggestion
---
title: Leah
type: authors
---
Leah is an information technology enthusiast. She is passionate about software development and technical writing. Her areas of interest are Data Science and Machine learning(ML). In addition, she loves playing computer games and adventure.
```

File: content/articles/using-multiple-ssh-keys-for-multiple-github-accounts/index.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+[Creating and updating the SSH config file](#creating-and-updating-the-ssh-config-file)
+[Cloning the Repositories](#cloning-the-repositories)
+
+### Prerequisites
+To grasp what this article entails. It is crucial to have a good understanding of how GitHub works.
+
+### What are SSH Keys
+[SSH refers to Secure Shell](https://en.wikipedia.org/wiki/Secure_Shell_Protocol). A cryptographic network protocol that allows a single computer to connect to a server over the internet securely. SSH is best used for accessing remote servers.
+
+SSH is designed to offer secure encryption, verification, and communication between computers. It provides a safe way of executing commands and configuring services remotely. The SSH technology is often performed using the client-server model. So you can securely log in remotely from one system to another and transmit files.
```
Comment (srishilesh) [inline content/articles/using-multiple-ssh-keys-for-multiple-github-accounts/index.md:17]: `client-server model`

Add 1 - 2 lines of explanation about it, and reference a link for more info about it.

File: content/articles/using-multiple-ssh-keys-for-multiple-github-accounts/index.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+Setting up SSH, you use a pair of keys, a public key and a private key. A public key is stored on the SSH server, and a private key is stored on your SSH client. So basically, SSH uses [asymmetric encryption](https://sectigostore.com/blog/what-is-asymmetric-encryption-how-does-it-work/#:~:text=Asymmetric%20encryption%20is%20a%20type,corresponding%20private%20key%20decrypts%20it.&text=The%20public%20key%20is%20open,and%20encrypt%20data%20with%20it.).
```
Comment (srishilesh) [inline content/articles/using-multiple-ssh-keys-for-multiple-github-accounts/index.md:19]:
```suggestion
Setting up SSH, you use a pair of keys, a public key and a private key. A public key is stored on the SSH server, and a private key is stored on your SSH client. So basically, SSH uses [asymmetric encryption](https://sectigostore.com/blog/what-is-asymmetric-encryption-how-does-it-work). 
```

Comment (srishilesh) [inline content/articles/using-multiple-ssh-keys-for-multiple-github-accounts/index.md:19]: More info about Private and Public keys would help

Comment (srishilesh) [reply content/articles/using-multiple-ssh-keys-for-multiple-github-accounts/index.md:19]: It's good that you have specified a referencing link. Adding a single of explanation would be better

File: content/articles/using-multiple-ssh-keys-for-multiple-github-accounts/index.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+Now that we have the SSH keys, let us link them with the Github account.
+To obtain the SSH key execute this command:
+
+`cat id_rsa_personal.pub`
+
+![Getting SSH key](/engineering-education/using-multiple-ssh-keys-for-multiple-github-accounts/ssh-copy-key.jpg)
+
+Copy the SSH key and then sign in to your GitHub account. Follow the steps below to add an SSH key to your GitHub account.
+1. On your GitHub, navigate to Settings
+2. Choose SSH and GPG keys
```
Comment (srishilesh) [inline content/articles/using-multiple-ssh-keys-for-multiple-github-accounts/index.md:75]: What is a GPG key?

File: content/articles/using-multiple-ssh-keys-for-multiple-github-accounts/index.md [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+#### Cloning personal repository
+Cloning your personal private projects, you use this command.
+`git clone git@github.com:your-github-account/private-project-repo.git`
+
+#### Cloning work repository
+Cloning your company's projects, you will use this command.
+`git clone git@github.myorganization.com:org-account/company-project-repo.git`
+
+### Conclusion
+That's all there is for this article. I hope it will be helpful. Also, I hope it helps you understand how to use multiple SSH keys for multiple GitHub accounts.
```
Comment (srishilesh) [inline content/articles/using-multiple-ssh-keys-for-multiple-github-accounts/index.md:127]: Give a brief summary of what you mentioned previously. Write key takeaways from this article. Tell how and where the reader could learn more about this.

If needed, you can add an additional `Further reading` section.

**Review by srishilesh — CHANGES_REQUESTED** [review]
@LeahKui 

Welcome to EngEd community!

Amazing article. It looks very beginner friendly.

I've suggested a few changes. Please let me know when they're done.

Thanks.

**Comment by LeahKui** [comment]
@srishilesh, thanks for reviewing this article. worked on the revisions

**Comment by ninjaginja** [comment]
plagiarism check completed - all good
[2120420-14fde1f5e2e7404e4898-1626987273357.pdf](https://github.com/section-engineering-education/engineering-education/files/6865161/2120420-14fde1f5e2e7404e4898-1626987273357.pdf)

**Review by srishilesh — APPROVED** [review]
@LeahKui 

Thanks for making the changes. This looks great!

Make sure to go through my commits, to understand and improve on your future articles.

Moving for the final review.

Thanks.

**Comment by louisefindlay23** [comment]
Hi, @LeahKui. Great tutorial. It certainly guides the users through each step. 

However, we value detailed and in-depth guides and feel you could add some detail to provide extra value to our readers. **Some suggested topics are:** What are SSH Keys? Why should you use SSH Keys with Git and GitHub? The difference between adding 2FA on your GitHub account vs using SSH Keys.

Thanks,
Louise - Content Moderator

**Comment by hectorkambow** [comment]
Overall great work @LeahKui 

cc @srishilesh

**Comment by LeahKui** [comment]
Thank you, @hectorkambow, @srishilesh, @louisefindlay23, for reviewing my article. I added the content @louisefindlay23 suggested

**Comment by louisefindlay23** [comment]
Thanks, @LeahKui for adding the extra content. I'm sure our readers will appreciate it. Final review complete :)

Thanks, @srishilesh for peer reviewing.

**Pull request merged**